Lib/argparse.py

```diff
@@ -2752,6 +2754,13 @@ def _get_formatter(self):
         formatter._set_color(self.color)
         return formatter
 
+    def _get_validation_formatter(self):
+        # Return cached formatter for read-only validation operations
+        # (_expand_help and _format_args). Avoids repeated _set_color calls.
```

Comment:
Since it's basically an optimizaton, you may add:

```suggestion
        # (_expand_help and _format_args). Avoids repeated slow _set_color calls.
```